File: backend/src/Domain/Entities/Expense.cs
using Domain.Enums;

namespace Domain.Entities;

public class Expense
{
    public Guid Id { get; set; }

    public string ExpenseNumber { get; set; } = null!;

    public Guid EmployeeId { get; set; }
    public Guid AttachmentId { get; set; }

    public DateOnly ExpenseDate { get; set; }

    public ExpenseCategory Category { get; set; }

    public decimal Amount { get; set; }
    public string Currency { get; set; } = null!;
    public string Description { get; set; } = null!;

    public ExpenseStatus Status { get; set; }

    public DateTime? SubmittedAt { get; set; }
    public DateTime? ApprovedAt { get; set; }
    public DateTime? ComplianceApprovedAt { get; set; }
    public DateTime? ReimbursedAt { get; set; }
    public DateTime? RejectedAt { get; set; }

    public Guid? ApprovedByEmployeeId { get; set; }
    public Guid? ComplianceApprovedByEmployeeId { get; set; }
    public Guid? ReimbursedByEmployeeId { get; set; }
    public Guid? RejectedByEmployeeId { get; set; }

    public string? RejectionComment { get; set; }

    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public Employee Employee { get; set; } = null!;
    public Attachment Attachment { get; set; } = null!;
}

File: frontend/src/types/expense.ts
export type ExpenseCategory =
  | 'Travel'
  | 'Hotel'
  | 'Meals'
  | 'OfficeSupplies'
  | 'ClientEntertainment'
  | 'Training'
  | 'Other'

export const EXPENSE_CATEGORIES: ExpenseCategory[] = [
  'Travel',
  'Hotel',
  'Meals',
  'OfficeSupplies',
  'ClientEntertainment',
  'Training',
  'Other',
]

export type ExpenseStatus =
  | 'Draft'
  | 'Submitted'
  | 'Approved'
  | 'ComplianceApproved'
  | 'Cancelled'
  | 'Reimbursed'
  | 'Rejected'

export const EXPENSE_STATUSES: ExpenseStatus[] = [
  'Draft',
  'Submitted',
  'Approved',
  'ComplianceApproved',
  'Cancelled',
  'Reimbursed',
  'Rejected',
]

export type ExpenseSortField =
  | 'expenseDate'
  | 'expenseNumber'
  | 'createdAt'
  | 'amount'
  | 'submittedAt'
  | 'approvedAt'
  | 'reimbursedAt'
  | 'rejectedAt'

export type ExpenseAction = 'Draft' | 'Submit'

export interface ExpenseResponse {
  id: string
  expenseNumber: string
  expenseDate: string
  category: ExpenseCategory
  amount: number
  currency: 'INR'
  description: string
  status: ExpenseStatus
  submittedAt: string | null
  approvedAt: string | null
  complianceApprovedAt: string | null
  rejectedAt: string | null
  rejectionComment: string | null
  reimbursedAt: string | null
  createdAt: string
  employeeName: string | null
  employeeNumber: string | null
  receiptAttachmentId: string
  attachmentOriginalFileName: string | null
}

export interface ExpenseEnvelopeResponse {
  expense: ExpenseResponse
}

export interface PagedExpenseResponse {
  items: ExpenseResponse[]
  page: number
  pageSize: number
  totalRecords: number
}
